File: intreverse.java
class Solution {
    public int reverse(int x) {
        int r = 0, y = x;
        while (y != 0) {
             if (r <Integer.MIN_VALUE / 10 || r > Integer.MAX_VALUE / 10) {
            return 0;
        }
        int digit = y % 10;
        y /= 10;
        r = r * 10 + digit;
        }    
        return r;
    }
}

File: intreverse.scala
object Solution {
    def reverse(x: Int): Int = {
        var y = x
        var r = 0
        while (y != 0) {
            val p = y % 10
            if ((r > 0 && r > (Int.MaxValue - y) / 10) || (r < 0 && r < (Int.MaxValue - y) / 10)) {
                return 0
            }
            r = r * 10 + p
            y /= 10
        }
        return r
    }
}
